### PR TITLE
Module support

### PR DESCRIPTION
**Pull request by zrbrown**
There's been discussion over the years of updating Testcontainers to work with Java modules (#1917, #1523, #1524), but not much movement on it as it's a breaking change. I'm submitting this PR to show the work that needs to be done to get this project working in a modular Java application.

First, split packages need to be fixed. Most modules were able to be fixed by just changed the `org.testcontainers.containers` part of the package to `org.testcontainers.<modulename>`. This did cause a few visibility issues, but I was surprised at how little package-private visibility was actually utilized! Besides package renaming, here are the other changes:

- `VncRecordingContainer::DEFAULT_RECORDING_FORMAT` visibility was changed from `package-private` to `public`
- `GenericContainer::getWaitStrategy` visibility was changed from `protected` to `public`
- `ContainerDatabaseDriver::getContainer` visibility was changed from `package-private` to `public`
- `TrinoContainer::DEFAULT_TAG` visibility was changed from `package-private` to `public`
- `PostgreSQLConnectionURLTest` tests were modified to allow for access to the protected method `constructUrlForConnection`

Second, each module's `build.gradle` was modified to add an automatic module name. I'm not super well-versed in Gradle, so there may be a way to generalize the jar task to reduce clutter in each module's build file.

Both of these changesets are independent, but in the spirit of addressing Java modules, they're related.

This is obviously a breaking change (though it's not _really_ that big of a change to consumers - all most will do is fix up imports), so it would need to target 2.0. I've seen discussion about a 2.0 (as well as a [milestone](https://github.com/testcontainers/testcontainers-java/milestone/20)), but as far as I can tell there isn't a branch to merge into. Since this project is used in tests, I personally wouldn't mind pulling in snapshot artifacts to use it if there were builds out there, but I can also see the headache in maintaining a branch where merges are more difficult due to package name changes. Thanks for any discussion on this!

### CODE REVIEW

**Comment by eddumelendez** [comment]
Hi @zrbrown, thanks for the PR! But, due to the number of conflicts and because this requires previous work as mentioned in [this comment](https://github.com/testcontainers/testcontainers-java/issues/7337#issuecomment-1657295307) I'm closing this PR.

**Comment by A248** [comment]
Dear @eddumelendez respectfully,

I speak to you in honesty when I ask if you considered your actions in closing this PR. It was submitted eight months ago after loads of effort and refactoring on behalf of its creator. Each item was addressed in particular, and modifications explained. The changes were motivated, the caveats listed, and relevancies noted, though without excessive detail. It was a model PR.

The submitter requested a review from your team. Authors were notified. The pull request waited in the queue. The maintainers did not submit reviews, or comments. Understandable: perhaps they were busy. The pull request stayed in this state for quite the while for eight months.

In the past 24 hours, you were the first maintainer to respond to this pull request. You closed this PR which, as the maintainer mentioned, was targeted for version 2.0 of test-containers, for which there does not exist a branch, by citing "a number of conflicts" between the PR and its target branch; furthermore by stating "this requires previous work," the content of which happens to be directly addressed by this PR.

In short, I hope that you have made an honest mistake, and I thank you for listening to your contributors.

**Comment by eddumelendez** [comment]
> I speak to you in honesty when I ask if you considered your actions in closing this PR. It was submitted eight months ago after loads of effort and refactoring on behalf of its creator.

We value every contribution that is made to the project. Because of that, we encourage contributors to discuss changes before to made them. We know there is time invested of those contributions and we don't want them to spend time on it if this not planned in short term.

`number of conflicts` is a tiny thing that I mentioned and there is other work coming before trying to merge this one and there will be even more conflicts and things to revisit. We will be happy to reopen this if it fits when we are aiming to achieve merging. If you wonder why not asking to update the PR, it is because we are not planning to do this soon.

Hope you understand the reasons.

**Comment by zrbrown** [comment]
Hopefully this PR did demonstrate how relatively easy (albeit slightly time-consuming) it is to add module support when the time comes to start work on 2.0!
